core/error.go

```diff
@@ -177,3 +177,20 @@ func IsValueTransferErr(e error) bool {
 	_, ok := e.(*ValueTransferError)
 	return ok
 }
+
+type BadHashErr struct {
+	Message string
+}
+
+func (err *BadHashErr) Error() string {
+	return err.Message
+}
+
+func BadHashError(hash common.Hash) error {
+	return &BadHashErr{Message: fmt.Sprintf("Found known bad hash in chain %x", hash)}
+}
+
+func IsBadHashErr(err error) bool {
+	_, ok := err.(*BadHashErr)
+	return ok
+}
```

Comment:
`os.IsNotExist` is an example for this technique.

Most of the errors in this file could just be variables instead
of a struct type containing the message and a constructor+predicate.